src/validation/joi/joi.shared.schemas.ts

```diff
@@ -174,3 +174,6 @@ export const baseDBEntitySchema: ObjectSchema<BaseDBEntity> = objectSchema<BaseD
 export const macAddressSchema = stringSchema.regex(/^([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})$/)
 
 export const uuidSchema = stringSchema.uuid()
+
+export const DATE_TIME_STRING_REGEX = /d{4}-\d{2}-\d{2}T\d{2}:\d{2}([+-]\d{2}:\d{2})?/
```

Comment:
I see that it's not generic enough, e.g it'd fail if seconds are included in the string (`2024-06-21T05:23:12`).

And it lacks a unit test with a set of inputs and outputs.

We'll need to decide how permissive we want to be here.
But, since we're "upgrading" from just stringSchema - almost anything is better (stricter) than stringSchema. So, probably "permissive" is a good approach: have optional seconds and optional timezone modifier.
Also, the timezone modifier of `Z` should be supported (as all unit tests use it).